src/Uno.UWP/Devices/Radios/Radio.Android.cs

```diff
@@ -0,0 +1,252 @@
+#if __ANDROID__
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+
+namespace Windows.Devices.Radios
+{
+
+	public partial class Radio
+	{
+
+		public RadioKind Kind { get; internal set; }
+
+		public string Name { get; internal set; }
+
+		public RadioState State { get; internal set; }
+
+		private static Radio GetRadiosBluetooth()
+		{
+			Android.Content.Context context = Android.App.Application.Context;
+			bool btFull = context.PackageManager.HasSystemFeature(Android.Content.PM.PackageManager.FeatureBluetooth);
+			bool btLE = context.PackageManager.HasSystemFeature(Android.Content.PM.PackageManager.FeatureBluetoothLe);
+			if (!(btFull || btLE))
+			{
+				return null;
+			}
+
+			var oRadio = new Radio();
+			oRadio.Kind = RadioKind.Bluetooth;
+			oRadio.Name = "Bluetooth";  // name as in UWP
+
+			if (!Windows.Extensions.PermissionsHelper.IsDeclaredInManifest(Android.Manifest.Permission.Bluetooth))
+			{
+				oRadio.State = RadioState.Unknown;
+				return oRadio;
+			}
+
+			if (Android.OS.Build.VERSION.SdkInt <= Android.OS.BuildVersionCodes.JellyBeanMr1)
+			{
+				// deprecated in API 31
+#pragma warning disable CS0618 // Type or member is obsolete
+				var btAdapter = Android.Bluetooth.BluetoothAdapter.DefaultAdapter;
+#pragma warning restore CS0618 // Type or member is obsolete
+				if (btAdapter == null)
+				{
+					return null;    // shouldn't happen...

```

Comment:
Why should it not happen? In all cases, use logging to explain to the user with debug information.

```
if(this.Log().IsEnabled(...Debug))
{
   this.Log().Debug("Could not get the bluetooth default adapter")

Reply:
`Radio does not contain a definition for Log()`

Reply:
And `this` cannot be used in `static` method. How to access Log inside `static`?